### PR TITLE
Feat/committee new

### PR DESCRIPTION
**Pull request by venkatkapil24**
Committee related features in a new branch

### CODE REVIEW

File: ipi/engine/forcefields.py [hunk 1]
```diff
@@ -1127,46 +1127,37 @@ def gather(self, r):
...
+                    "s": list(rescaled_virs[i_r].flatten()),
+                    "x": xtrs[i_r],
+                }
+            )
+            
+        r["result"][3] = {
+            "committee_pot" : rescaled_pots,
+            "committee_force" : rescaled_frcs.reshape(len(rescaled_pots),-1),
+            "committee_virial" : rescaled_virs.reshape(len(rescaled_pots),-1),
+            }
```
Comment (venkatkapil24) [inline ipi/engine/forcefields.py]: shall we also add a "committee_weight" ? May be beneficial for a user to have direct access to it.

File: ipi/engine/forcefields.py [hunk 2]
```diff
@@ -1118,6 +1118,7 @@ def gather(self, r):
             "committee_pot" : rescaled_pots,
             "committee_force" : rescaled_frcs.reshape(len(rescaled_pots),-1),
             "committee_virial" : rescaled_virs.reshape(len(rescaled_pots),-1),
+            "raw" : ""
```
Comment (venkatkapil24) [inline ipi/engine/forcefields.py]: Shouldn't raw print out everything ?

File: examples/lammps/h2o-committee/init.pdb [hunk 3]
```diff
@@ -0,0 +1,651 @@
+TITLE cell{atomic_unit} positions{atomic_unit}
```
Comment (mahrossi) [inline examples/lammps/h2o-committee/init.pdb:1]: IMHO we should avoid giving examples with pdb files as inputs. but it is not a big deal. it is just too easy to make a mistake with them and the i-PI parser for pdb is not the most robust in the world.

File: ipi/engine/forcefields.py [hunk 4]
```diff
@@ -885,3 +911,237 @@ def evaluate(self, r):
...
+        )
+        rescaled_virs = np.asarray(
+            [mean_vir + self.alpha * (vir - mean_vir) for vir in virs]
+        )
+
+        # Calculates the error associated with the committee
+        var_pot = np.var(rescaled_pots, ddof=1)
+        std_pot = np.sqrt(var_pot)
+
+        if self.baseline_name != "":
```
Comment (mahrossi) [inline ipi/engine/forcefields.py:1057]: Maybe it is difficult for a reason I cannot immediately see, but the code below looks like something that would be much better encapsulated in one or two functions?

File: ipi/inputs/forcefields.py [hunk 5]
```diff
@@ -614,3 +616,147 @@ def fetch(self):
             dopbc=self.pbc.fetch(),
             threaded=self.threaded.fetch(),
         )
+
+
+class InputFFCommittee(InputForceField):
+    default_help = """A class to consolidate multiple FF inputs in a single XML field.
```
Comment (mahrossi) [inline ipi/inputs/forcefields.py:622]: It sounds completely cryptic to me this first phrase -- no idea what it means to consolidate FF inputs into a XML field. And the phrase implies one can use an XML field for uncertainty estimation (which I don't think was the exact idea). More helpful description, please?

File: tools/py/committee-reweight.py [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+import sys
+import argparse
+import numpy as np
+from ipi.inputs.simulation import InputSimulation
+from ipi.utils.io.inputs import io_xml
+
+
+def direct_reweight(pot, obs, kbT):
+    """
+    reweights an observable based on a committee of potentials, at the temperature kT
```
Comment (mahrossi) [inline tools/py/committee-reweight.py:11]: any reference to give here, how the reweighting is done in practice?

File: tools/py/committee-reweight.py [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+def CEA(pot, obs, kbT):
+    """
+    reweight the quantity by cumulant expansion approximation (CEA)
+    ref:
+    Michele Ceriotti,  Guy A. R. Brain, Oliver Riordan and David E.
+    Manolopoulos 2011The inefficiency of re-weighted sampling and
+    the curse of system size in high-order path integration
+    Proc. R. Soc. A.4682–17
+    http://doi.org/10.1098/rspa.2011.0413
+    :return:
```
Comment (mahrossi) [inline tools/py/committee-reweight.py:51]: missing? return what?

File: tools/py/committee-reweight.py [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+    elif stride < 0:
+        stride = np.abs(stride)
+        obs = obs[::stride]
+    else:
+        raise ValueError(
+                "Stride value cannot be zero"
+                )
+
+
+    # Load kbT from i-PI, we could make it into a small function
```
Comment (mahrossi) [inline tools/py/committee-reweight.py:121]: does this handle different possible units for T and so on? I am assuming yes already in xml_parse. Comment here to check.

File: tools/py/committee-reweight.py [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+    else:
+        rw_obs, _h_matrix = CEA(potentials, obs, kbT)
+
+    np.savetxt("rw_" + obs_file, rw_obs)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="""This tool exploits a committee of ML models to compute the reweighted  canonical averages of a given observable, in order to quantify its uncertainty at thermodynamic equilibrium.
+    The units are assumed to be atomic, which are also the internal units of i-PI.
```
Comment (mahrossi) [inline tools/py/committee-reweight.py:143]: hm! why not call the unit conversion functions of i-PI for more flexibility, etc.?

File: examples/lammps/h2o-committee/init.pdb (hunk 3 above)
Comment (ceriottm) [reply examples/lammps/h2o-committee/init.pdb:1]: I agree, but pretty much all water examples in lammps/ use this init.pdb. 
I'll open an issue so we can get rid of these (and actually perhaps use a single water.xyz so we save space?

Comment (ceriottm) [reply examples/lammps/h2o-committee/init.pdb:1]: Mentioned in #173

File: ipi/engine/forcefields.py (hunk 4 above)
Comment (ceriottm) [reply ipi/engine/forcefields.py:1057]: it's not used anywhere else... how would you split it up in a way in which the function simplified the code more than by moving this rather lengthy block to a different place?

File: tools/py/committee-reweight.py (hunk 9 above)
Comment (ceriottm) [reply tools/py/committee-reweight.py:143]: this only really matters for pot, and we have no way of outputting committee pots as anything but atomic units. agree the text could be clearer

File: tools/py/committee-reweight.py (hunk 8 above)
Comment (giulioi) [reply tools/py/committee-reweight.py:134]: Yes, the idea is that by pulling the temperature from the i-PI parser it takes care of the units of the temperature. The only thing to care about is feeding energies in Ha.

File: tools/py/committee-reweight.py (hunk 7 above)
Comment (giulioi) [reply tools/py/committee-reweight.py]: Added proper indication of the inputs and outputs of all of the functions. Tried to keep a uniform style.

File: tools/py/committee-reweight.py (hunk 6 above)
Comment (giulioi) [reply tools/py/committee-reweight.py]: Added a reference for the direct reweight scheme

File: tools/py/committee-reweight.py (hunk 9 above)
Comment (giulioi) [reply tools/py/committee-reweight.py]: Made clearer that the potential is expected to be in Hartree. This is meant as a warning for people who want to use this tool after running simulations with a different code

File: ipi/engine/forcefields.py (hunk 4 above)
Comment (mahrossi) [reply ipi/engine/forcefields.py:1057]: ok, sometimes things like the uncertainty calls (there are at least 2 below right?) could be useful in many other parts of the code in the future, but sure. It is not hard to rewrite.

**Comment by mahrossi** [comment]
Just to document the issues found: There was a "bug" in the regtest infrastructure when two different drivers were used (runstools.py routine). @eszterpos commited a first fix so that the regression test for this branch would work ( f3c505b ). It does not seems to be the end of the story though.

**Comment by ceriottm** [comment]
Looks like it works, it's decently documented and all the tests pass. 
Merging so we can focus on the regtests.

**Pull request merged**